### DISCUSSION TITLE
Do we have a way to know which data is stale or new?

### DISCUSSION THREAD

**Discussion by anothertempore** [vercel/swr#782] [Q&A]
Hi guys!

I have a scenario that needs totally rely on the data responded from the API. SWR always returns the cache data first then revalidate it in the background. If I get data I'll show some important messages to users. If I get the cached data first, this message will be shown, which might not the correct message.

So I'm thinking if I can know what kind of data is a staled data, I can ignore them in this scenario. 🤔

Thanks!

**Comment by shuding** [comment]
Hi! First I have a question regarding the scenario that you are dealing with: if we fetch the API and the response is the same as our cached data, should we show the message?

**Reply by anothertempore** [reply]
Yes we should!

**Comment by mickomagallanes** [comment]
Is this already solved? I need something like this
